src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,37 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const status = useAppSelector(state => state.filter.status);
+  const query = useAppSelector(state => state.filter.query);
+
+  const handleStatus = (value: Status) => {
+    dispatch(filterActions.status(value));
+  };
+
+  const handleQuery = (value: string) => {
+    if (!value.trim()) {
+      dispatch(filterActions.query(''));
+    } else {
+      dispatch(filterActions.query(value));
+    }
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={event => handleStatus(event.target.value as Status)}
```

Comment:
```suggestion
            onChange={handleStatus}
```
Dirty trick. What if you mispell option value or decide to remove one option from reducer, but don't remove here? Better solution is:
1) set option values from enum (Status.ALL, ...);
2) accept event as handler argument;
3) get target value and switch it comparing to Status enum values.